docs/proposed/http-storage-node-protocol.rst

```diff
@@ -0,0 +1,506 @@
+.. -*- coding: utf-8 -*-
+
+Storage Node Protocol ("Great Black Swamp", "GBS")
+==================================================
+
+The target audience for this document is Tahoe-LAFS developers.
+After reading this document,
+one should expect to understand how Tahoe-LAFS clients interact over the network with Tahoe-LAFS storage nodes.
+
+The primary goal of the introduction of this protocol is to simplify the task of implementing a Tahoe-LAFS storage server.
+Specifically, it should be possible to implement a Tahoe-LAFS storage server without a Foolscap implementation
+(substituting a simpler GBS server implementation).
+The Tahoe-LAFS client will also need to change but it is not expected that it will be noticably simplified by this change
+(though this may be the first step towards simplifying it).
+
```

Comment:
+5 I like the way this document introduces the concepts with a good overview (this part included) :+1: